File: server/datasets/java_source-code/Multiple-B2/logic_substituted_1.java
// Two Sum (hash map) - DISGUISE: logic-substituted
// Derived from unique_1.py. The membership check "if complement in seen"
// has been rewritten as its boolean equivalent "!(!seen.containsKey(complement))".

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int index = 0; index < nums.length; index++) {
            int value = nums[index];
            int complement = target - value;
            if (!(!seen.containsKey(complement))) {
                return new int[]{seen.get(complement), index};
            }
            seen.put(value, index);
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] result = twoSum(nums, target);
        System.out.println("Two Sum result: " + Arrays.toString(result));
    }
}

File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_b2/logic_substituted_1.py
# Two Sum (hash map) - DISGUISE: logic-substituted
# Derived from unique_1.py. The membership check "if complement in seen"
# has been rewritten as its De Morgan's-law equivalent
# "if not (complement not in seen)" - logically identical, syntactically
# different comparison, matching the "swap an equivalent boolean/math
# expression" disguise family.

def two_sum(nums, target):
    seen = {}
    for index, value in enumerate(nums):
        complement = target - value
        if not (complement not in seen):
            return [seen[complement], index]
        seen[value] = index
    return None


def main():
    nums = [2, 7, 11, 15]
    target = 9
    result = two_sum(nums, target)
    print("Two Sum result:", result)


if __name__ == "__main__":
    main()
